index.js

```diff
@@ -1,97 +1,98 @@
-
-    const div = document.createElement('div');
-    div.className = 'book-list';
-
-    display.appendChild(div);
-
-    const title = document.createElement('p');
-    title.textContent = book.title;
-    title.className = 'class-title';
-
-    div.appendChild(title);
-
-    const author = document.createElement('p');
-    author.textContent = book.author;
-    author.className = 'class-title';
-
-    div.appendChild(author);
-
-    // const bookItem = localStorage.setItem('book', JSON.stringify(book));
-
-    const remove = document.createElement('button');
-    remove.textContent = 'Remove';
-    remove.className = 'remove';
 
-    div.appendChild(remove);
-
-    remove.addEventListener('click', (e) => {
-      removeBook(e.target);
-      e.preventDefault();
-    });
+  add(book) {
+    this.books.push(book);
   }
-};
 
-// create book object and add it to array and local store
-const addBook = () => {
-  const bookTitle = document.querySelector('.title');
-  const bookAuthor = document.querySelector('.author');
-  const newBook = {};
-  newBook.title = bookTitle.value;
-  newBook.author = bookAuthor.value;
-
-  // add book to the array
-  book.push(newBook);
+  remove(book) {
+    this.books = this.books.filter((i) => i !== book);
+  }
 
-  // save it to local storage
-  saveData(book);
+  get(index) {
+    return this.books[index];
+  }
+}
 
-  // display the book on the page
-  displayBook(newBook);
+// eslint-disable-next-line max-classes-per-file
+class Book {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+  }
+}
 
-  // clear fields
-  bookTitle.value = '';
-  bookAuthor.value = '';
-};
+const mBooks = [];
+const myCollection = new Collection();
 
-// fired on page load to show data from the local store
-const showSavedBook = () => {
-  // const books = localStorage.getItem('savedBooks');
-  const books = JSON.parse(localStorage.getItem('savedBooks'));
-  if (books) {
-    books.forEach(displayBook);
+function removeBook(newBook) {
+  newBook.forEach((b) => {
+    const display = document.getElementById('book-list');
+    const div = document.createElement('div');
+    div.classList = 'book';
+    div.innerHTML = `<div>"${b.title}" by ${b.author}</div><div><button class="remove">Remove</button></div>`;
+    display.appendChild(div);
+    const deletebtn = document.querySelectorAll('.remove');
+
+    deletebtn.forEach((btn) => {
+      btn.addEventListener('click', (e) => {
+        const book = e.target.parentElement.parentElement;
+
+        mBooks.forEach((b) => {
+          if (`"${b.title}" by ${b.author}` === book.firstElementChild.innerHTML) {
+            const index = mBooks.indexOf(b);
+            mBooks.splice(index, 1);
+            localStorage.setItem('myCollection', JSON.stringify(mBooks));
+          }
+        });
+
+        myCollection.remove(book);
+        book.remove();
+        e.preventDefault();
+      });
+    });
```

Comment:
Instead of creating a function for removing a book, Please create a class method to remove a book from the Book Store 👍🏻